### DISCUSSION TITLE
Which option to go with? tKey (self host SDK) or custom verifier?

### DISCUSSION THREAD

**Discussion by auracheeze** [Web3Auth/Web3Auth#1162] [Help]
My objective is to be able to create the same wallet/key for a user regardless of whether the user signs in with her Google account or Twitter account.
I read that an aggregate verifier is needed (https://web3auth.io/docs/custom-authentication/verifiers#aggregate-verifiers).
At the same time, the self-host option (https://web3auth.io/docs/sdk/self-host/) seemingly can achieve this objective as well.
Are these two options both viable to achieve this goal?

**Comment by shahbaz17** [comment]
Hey @auracheeze 

If your objective is to create the same wallet/key for a user regardless of login provider, provided they both use the same email, or there must be a unique identifier between the JWT returned from each provider. Let's say `email`. 

Then, you can aggregate the verifiers and get the same key back.

The question here is whether to choose aggregate verifiers or self-host. So, the aggregate verifier is a feature and can be available to both the Self-Host SDK and our Plug n Play SDKs.
